ui/models/job.js

```diff
@@ -118,35 +140,39 @@ export default class JobModel {
     try {
       notify(`Attempting to retrigger ${jobTerm} via actions.json`, 'info');
 
-      /* eslint-disable no-await-in-loop */
-      for (const id of jobIds) {
-        const job = await JobModel.get(repoName, id);
-        const decisionTaskId = await getGeckoDecisionTaskId(
-          job.push_id,
-          repoName,
-        );
-        const results = await TaskclusterModel.load(decisionTaskId, job);
-        const retriggerTask = results.actions.find(
-          result => result.name === 'retrigger',
-        );
-
-        try {
-          await TaskclusterModel.submit({
-            action: retriggerTask,
-            decisionTaskId,
-            taskId: results.originalTaskId,
-            input: {},
-            staticActionVariables: results.staticActionVariables,
-          });
-        } catch (e) {
-          // The full message is too large to fit in a Treeherder
-          // notification box.
-          notify(formatTaskclusterError(e), 'danger', { sticky: true });
+      const uniquePerPushJobs = {};
+      const jobs = await JobModel.get(repoName, jobIds);
+      for (const job of jobs) {
+        if (!uniquePerPushJobs[job.push_id]) {
+          uniquePerPushJobs[job.push_id] = [];
         }
+        uniquePerPushJobs[job.push_id].push(job.job_type_name);
       }
-      /* eslint-enable no-await-in-loop */
 
-      notify(`Request sent to retrigger ${jobTerm} via action.json`, 'success');
+      for (const [key, value] of Object.entries(uniquePerPushJobs)) {
+        getGeckoDecisionTaskId(Number(key), repoName).then(decisionTaskId => {
+          TaskclusterModel.load(decisionTaskId).then(async results => {
+            const actionTaskId = slugid();
+            const addNewJobsTask = results.actions.find(
+              action => action.name === 'add-new-jobs',
+            );
+
+            await TaskclusterModel.submit({
+              action: addNewJobsTask,
+              actionTaskId,
+              decisionTaskId,
+              taskId: null,
```

Comment:
Here's where @djmitche has much more domain knowledge than I do.  :)  But I was surprised we don't need the ``originalTaskId`` and can just use ``input: { tasks: job.job_type_name }`` here.  I don't have the scopes to actually test this, either.  So I'm going to trust Dustin that this is the right way to go.

Reply:
I blatantly ~~stole~~**borrowed** this from https://github.com/mozilla/treeherder/blob/7a48bbc575863675e73f1c7f82f4509fd778961a/ui/models/push.js#L181-L188

Reply:
It's because this action is a per-push action, rather than a per-task action.